usr/src/cmd/fm/modules/common/zfs-diagnosis/zfs_de.c

```diff
@@ -796,6 +827,39 @@ zfs_fm_recv(fmd_hdl_t *hdl, fmd_event_t *ep, nvlist_t *nvl, const char *class)
 	if (fmd_case_solved(hdl, zcp->zc_case))
 		return;
 
+	zhdl = fmd_hdl_getspecific(hdl);
+
+	/*
+	 * Find the corresponding pool.
+	 */
+	cb.cb_guid = pool_guid;
+	cb.cb_zhp = NULL;
+	if (zhdl != NULL && zpool_iter(zhdl, find_pool, &cb) != 0)
+		return;
+
+	zhp = cb.cb_zhp; /* NULL if pool was not found. */
+
+	/*
+	 * The libzfs API takes a string representation of a base-10 guid here
+	 * instead of a number, likely because the primary libzfs consumer are
+	 * the CLI tools.
+	 */
+	vdg[vdg_size - 1] = '\0';
+	vp = ulltostr(vdev_guid, &vdg[vdg_size - 1]);
```

Comment:
I would just use snprintf here - it's also more portable anyways (ulltostr is a SVR4-ism)